backend/services/project_search_service.py

```diff
@@ -268,7 +268,10 @@ def _filter_projects(search_dto: ProjectSearchDTO, user):
                 for mapping_type in search_dto.mapping_types
             ]
 
-            query = query.filter(Project.mapping_types.contains(mapping_type_array))
+            if search_dto.mapping_types_exact and len(mapping_type_array) == 1:
```

Comment:
I don't like the previous implementation with contains and that limitation of only one `mappingType` on the exact matching, because it is not intuitive to the users (we allow users to search for more than one type) and it's quite restrictive.

For example, I would like to get all projects to map buildings or roads, I was expecting to receive projects:
- with only buildings
- with only roads
- with only these two types
- with these two types and others

or projects with only the two types on the exact match search.

If the implementation is too difficult or expensive to the database, we can keep that way. Do you have an opinion, @ramyaragupathy?

Reply:
Let me see the changes I can make to get this working in that format.